src/test/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResourceTest.java

```diff
@@ -433,87 +438,212 @@ public void testAddTags() throws Exception {
+            stats = createImageStat("other");
+            json = toJsonString(stats);
+            body = given().contentType(ContentType.JSON).header("token", token).body(json).when()
+                    .post(StatsTestHelper.BASE_URL + "?t=" + myTag).body();
+            result = body.as(ImageStats.class);
+            assertEquals(myTag, result.getTag());
+            assertTrue(result.getId() > 0);
+            statIds.add(result.getId());
+
+            // Add a third one without a tag
+            stats = createImageStat("third");
+            json = toJsonString(stats);
+            body = given().contentType(ContentType.JSON).header("token", token).body(json).when()
+                    .post(StatsTestHelper.BASE_URL).body();
+            result = body.as(ImageStats.class);
+            assertNull(result.getTag());
+            assertTrue(result.getId() > 0);
+            statIds.add(result.getId());
+
+            // Find stats by tag
+            ImageStats[] results = given().when().contentType(ContentType.JSON).header("token", token)
+                    .get(StatsTestHelper.BASE_URL + "/tag/" + myTag).body().as(ImageStats[].class);
+            assertEquals(2, results.length);
+            for (ImageStats s : results) {
+                assertEquals(myTag, s.getTag());
+                assertTrue("other".equals(s.getImageName()) || "foo-stat".equals(s.getImageName()));
+            }
+        } finally {
+            // Delete them again
+            String imageIdsJson = toJsonString(statIds.toArray(new Long[0]));
+            ImageStats[] deletedIds = given().contentType(ContentType.JSON).header("token", token).body(imageIdsJson).when()
+                    .delete(StatsTestHelper.BASE_URL).body().as(ImageStats[].class);
+            assertEquals(3, deletedIds.length);
+
+            // no more image stats
+            given().when().header("token", token).get(StatsTestHelper.BASE_URL).then().statusCode(200).body(is("[]"));
+            TestUtil.checkLog();
         }
+    }
 
-        // Delete them again
-        String imageIdsJson = toJsonString(statIds.toArray(new Long[0]));
-        ImageStats[] deletedIds = given().contentType(ContentType.JSON).header("token", token).body(imageIdsJson).when()
-                .delete(StatsTestHelper.BASE_URL).body().as(ImageStats[].class);
-        assertEquals(3, deletedIds.length);
+    @Test
+    public void testGetStats() throws Exception {
+        String token = StatsTestHelper.login(Mode.READ_WRITE);
+        String myTag = "some-run";
+        List<Long> statIds = new ArrayList<>();
 
-        // no more image stats
-        given().when().header("token", token).get(StatsTestHelper.BASE_URL).then().statusCode(200).body(is("[]"));
-        TestUtil.checkLog();
+        try {
+            ImageStats stats = createImageStat("foo-stat", myTag);
+            stats.setCreatedAt(Date.from(LocalDateTime.now().minusDays(10).toInstant(java.time.ZoneOffset.UTC)));
```

Comment:
Setting this doesn't work for testing since it gets overridden in https://github.com/Karm/collector/blob/fb65ceb748cefe4b0c9b6cd02f1a76c22682928f/src/main/java/com/redhat/quarkus/mandrel/collector/report/model/ImageStatsCollection.java#L68 any ideas how I could work around it?

Reply:
`import.sql` in resources? See: https://quarkus.io/guides/hibernate-orm#quarkus-hibernate-orm_quarkus-hibernate-orm-sql-load-script

Reply:
I decided the simplest thing to do was to allow the client to explicitly set the creation date. After all we might want to retroactively add some entries.

https://github.com/Karm/collector/pull/35/commits/d17b9144e77cdde713df384508c2189222a9fdf3

Reply:
Fair enough.